src/createServer.js

```diff
@@ -1,3 +1,55 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+
+const { convertToCase } = require('./convertToCase');
+
+const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const parsedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const textToConvert = parsedUrl.pathname.slice(1);
+    const toCase = parsedUrl.searchParams.get('toCase');
+    const errors = [];
+
+    if (!textToConvert) {
+      errors.push(
+        { message: 'Text to convert is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' },
+      );
+    }
+
+    if (!toCase) {
+      errors.push(
+        { message: '"toCase" query param is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' },
+      );
+    } else if (!supportedCases.includes(toCase)) {
+      errors.push(
+        { message: 'This case is not supported. '
+        + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.' },
+      );
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const result = convertToCase(textToConvert, toCase);
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+
+    res.end(JSON.stringify({
```

Comment:
`writeHead` returns res, so this can be chained
```suggestion
    res.writeHead(200, { 'Content-Type': 'application/json' })
      .end(JSON.stringify({
```